app/src/main/java/org/simple/clinic/bloodsugar/entry/BloodSugarEntrySheet.kt

```diff
@@ -306,22 +291,30 @@ class BloodSugarEntrySheet : BottomSheetActivity(), BloodSugarEntryUi, RemoveBlo
     bloodSugarDateButton.text = dateFormatter.format(date)
   }
 
-  override fun showEditRadomBloodSugarTitle() {
-    enterBloodSugarTitleTextView.visibility = View.GONE
-    editBloodSugarTitleTextView.visibility = View.VISIBLE
-    editBloodSugarTitleTextView.text = getString(R.string.bloodsugarentry_edit_random_title)
-  }
+  override fun showEntryTitle(measurementType: BloodSugarMeasurementType) {
+    enterBloodSugarTitleTextView.visibility = View.VISIBLE
+    editBloodSugarTitleTextView.visibility = View.GONE
 
-  override fun showEditPostPrandialBloodSugarTitle() {
-    enterBloodSugarTitleTextView.visibility = View.GONE
-    editBloodSugarTitleTextView.visibility = View.VISIBLE
-    editBloodSugarTitleTextView.text = getString(R.string.bloodsugarentry_edit_post_prandial_title)
+    val title = when (measurementType) {
+      Random -> getString(R.string.bloodsugarentry_random_title)
+      PostPrandial -> getString(R.string.bloodsugarentry_post_prandial_title)
+      Fasting -> getString(R.string.bloodsugarentry_fasting_title)
+      else -> throw AssertionError("Unknown measurement type $measurementType")
```

Comment:
I am not sure if this should throw an `Assertion error`. I think we should display the value we have received in the `Unknown` type. We added `Unknown` for forward compatibility. 
Replace `else` with the following:
```
is Unknown -> measurementType.actualValue
```